stubs/python-xlib/Xlib/protocol/rq.pyi

```diff
@@ -0,0 +1,416 @@
+    structcode: None
+    def pack_value(self, val: str) -> bytes: ...
+    def parse_binary(self, data: bytes | bytearray, display: Unused) -> tuple[str, bytes | bytearray]: ...
+
+Str: StrClass
+
+class Struct:
+    name: str
+    check_value: Callable[[Any], Any] | None
+    keyword_args: bool
+    fields: tuple[Field]
+    static_codes: str
+    static_values: int
+    static_fields: list[Field]
+    static_size: int
+    var_fields: list[Field]
+    structcode: str | None
+    structvalues: int
+    def __init__(self, *fields: Field) -> None: ...
+    def to_binary(self, *varargs: object, **keys: object) -> bytes: ...
+    def pack_value(self, value: tuple[object, ...] | dict[str, object] | DictWrapper) -> bytes: ...
+    @overload
+    def parse_value(self, val: SliceableBuffer, display: display.Display | None, rawdict: Literal[True]) -> dict[str, Any]: ...
+    @overload
+    def parse_value(
+        self, val: SliceableBuffer, display: display.Display | None, rawdict: Literal[False] = ...
+    ) -> DictWrapper: ...
+    @overload
+    def parse_binary(
+        self, data: SliceableBuffer, display: display.Display | None, rawdict: Literal[True]
+    ) -> tuple[dict[str, Any], SliceableBuffer]: ...
+    @overload
+    def parse_binary(
+        self, data: SliceableBuffer, display: display.Display | None, rawdict: Literal[False] = ...
+    ) -> tuple[DictWrapper, SliceableBuffer]: ...
+    # Structs generate their attributes
+    # TODO: Complete all classes inheriting from Struct
+    # and create a type-only class for all direct instances
+    @type_check_only
+    def __getattr__(self, __name: str) -> Any: ...
+
+class TextElements8(ValueField):
+    string_textitem: Struct
+    def pack_value(  # type: ignore[override]  # Override Callable
+        self, value: Iterable[Field | str | bytes | tuple[Sequence[object], ...] | dict[str, Sequence[object]] | DictWrapper]
+    ) -> tuple[bytes, None, None]: ...
+    def parse_binary_value(  # type: ignore[override]  # See: https://github.com/python-xlib/python-xlib/pull/249
+        self, data: SliceableBuffer, display: display.Display | None, length: Unused, format: Unused
+    ) -> tuple[list[DictWrapper], Literal[""]]: ...
+
+class TextElements16(TextElements8):
+    string_textitem: Struct
+
+class GetAttrData:
+    # GetAttrData classes get their attributes dynamically
+    # TODO: Complete all classes inheriting from GetAttrData
+    def __getattr__(self, attr: str) -> Any: ...
+    @type_check_only
+    def __setattr__(self, __name: str, __value: Any) -> None: ...
+
+class DictWrapper(GetAttrData):
+    def __init__(self, dict: dict[str, object]) -> None: ...
+    def __getitem__(self, key: str) -> object: ...
+    def __setitem__(self, key: str, value: object) -> None: ...
+    def __delitem__(self, key: str) -> None: ...
+    def __setattr__(self, key: str, value: object) -> None: ...
+    def __lt__(self, other: object) -> bool: ...
+    def __gt__(self, other: object) -> bool: ...
```

Comment:
> The runtime implementation of these methods confuses me. If this class is a wrapper around a dict, then surely these methods will always fail??

I mean.... yes? XD
```
>>> from Xlib.protocol.rq import DictWrapper
>>> DictWrapper({}) < DictWrapper({})
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/mnt/e/Users/Avasam/Documents/Git/python-xlib/Xlib/protocol/rq.py", line 1480, in __lt__    else:
TypeError: '<' not supported between instances of 'dict' and 'dict'
```

Seems it was naïvely adapted from python2's `cmp` (which did work with dicts)
https://github.com/python-xlib/python-xlib/commit/39b27b31661497d9b12d828543d8afe9bf8e2f81

Reply:
Haha, amazing 😅

Maybe we'd be doing users a favour if we left these methods out of the stub, since they'll just unconditionally fail if they're ever called? We can add an allowlist entry, with a comment to explain why they're deliberately omitted?

Reply:
Sounds like the way to go. Stubsabot/stubtest should pick it up if it's fixed (I've raised an issue in the lib, idk how actively, but it's still maintained)